In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib
import matplotlib.pyplot as plt
import os
import re
from itertools import compress
from nltk.tokenize import PunktSentenceTokenizer

# Debatepedia Preprocessing

## Import Data

In [2]:
df0 = pd.read_csv("/home/weci2587/projects/wstud-visit-the-dome-ss19/data/debatepedia/labeled_frames_args_merged_reindexed.csv", sep = "|")
print(len(df0))

12326


## Preprocessing Conclusions

### Remove Duplicate Conclusions

In [3]:
duplicateConclusions = []
index_i = 0
for conclusion in df0["conclusion"]:
    index_j = 0
    check = False
    for duplicateConclusion in df0["conclusion"]:
        if conclusion == duplicateConclusion and index_i != index_j and  not index_j < index_i:
            if not check:
                print("Duplicate at " + str(index_i) + "  ---  " + conclusion)
                check = True
            print("Duplicate at " + str(index_j) + "  ---  " + duplicateConclusion)
            duplicateConclusions.append(index_j)
        index_j += 1
    if check:
        print("")
    index_i += 1

Duplicate at 194  ---  Nuclear weapons make the world less secure
Duplicate at 483  ---  Nuclear weapons make the world less secure

Duplicate at 196  ---  Nuclear submarine accidents are a global hazard
Duplicate at 484  ---  Nuclear submarine accidents are a global hazard

Duplicate at 202  ---  US nuclear weapons reduce the intensity of international conflicts
Duplicate at 486  ---  US nuclear weapons reduce the intensity of international conflicts

Duplicate at 223  ---  Abortion may be immoral, but it is still a woman's right
Duplicate at 225  ---  Abortion may be immoral, but it is still a woman's right

Duplicate at 250  ---  No one argues that an acorn or even an animal fetus has a “sanctity” or rights.
Duplicate at 267  ---  No one argues that an acorn or even an animal fetus has a “sanctity” or rights.

Duplicate at 404  ---  Non-circumcision can result in many complications
Duplicate at 6474  ---  Non-circumcision can result in many complications

Duplicate at 405  ---  The 

Duplicate at 670  ---  It is perfectly natural to use animals for our own nutrition and pleasure.
Duplicate at 6158  ---  It is perfectly natural to use animals for our own nutrition and pleasure.

Duplicate at 672  ---  Animals cannot possess rights because they have no moral judgement
Duplicate at 6192  ---  Animals cannot possess rights because they have no moral judgement
Duplicate at 11781  ---  Animals cannot possess rights because they have no moral judgement

Duplicate at 673  ---  Animals cannot make moral claims so cannot claim rights
Duplicate at 6193  ---  Animals cannot make moral claims so cannot claim rights
Duplicate at 11782  ---  Animals cannot make moral claims so cannot claim rights

Duplicate at 674  ---  That the retarded have rights does not justify animal rights
Duplicate at 6194  ---  That the retarded have rights does not justify animal rights
Duplicate at 11784  ---  That the retarded have rights does not justify animal rights

Duplicate at 675  ---  Animal r

Duplicate at 1675  ---  Allowing public nudity risks exposure to children
Duplicate at 8993  ---  Allowing public nudity risks exposure to children

Duplicate at 1676  ---  Public nudity is a valuable expression of freedom.
Duplicate at 8991  ---  Public nudity is a valuable expression of freedom.

Duplicate at 1677  ---  Banning public nudity is offensive to nudists.
Duplicate at 8992  ---  Banning public nudity is offensive to nudists.

Duplicate at 1690  ---  Public smoking ban protects health of non-smokers.
Duplicate at 10491  ---  Public smoking ban protects health of non-smokers.

Duplicate at 1691  ---  Public smoking ban protects rights of non-smokers.
Duplicate at 10492  ---  Public smoking ban protects rights of non-smokers.

Duplicate at 1692  ---  Smokers can/should smoke in private places, not in public.
Duplicate at 10493  ---  Smokers can/should smoke in private places, not in public.

Duplicate at 1734  ---  Trans fats extend shelf life of foods.
Duplicate at 1761  ---

Duplicate at 2575  ---  Civil unions are "separate, but not equal"
Duplicate at 5308  ---  Civil unions are "separate, but not equal"

Duplicate at 2627  ---  Obama has a strong record of legislative achievement
Duplicate at 7940  ---  Obama has a strong record of legislative achievement

Duplicate at 2631  ---  Obama would be a fresh, uncorrupted face in the White House
Duplicate at 7939  ---  Obama would be a fresh, uncorrupted face in the White House

Duplicate at 2632  ---  Obama's diverse global background is valuable.
Duplicate at 7938  ---  Obama's diverse global background is valuable.

Duplicate at 2637  ---  Obama inspires people to become better citizens.
Duplicate at 7963  ---  Obama inspires people to become better citizens.

Duplicate at 2711  ---  Pesticide use.
Duplicate at 2712  ---  Pesticide use.

Duplicate at 2819  ---  Liberal democracies should ensure a degree of "equality of outcome"
Duplicate at 7576  ---  Liberal democracies should ensure a degree of "equality 

Duplicate at 4133  ---  School vouchers erode national educational standards.
Duplicate at 4145  ---  School vouchers erode national educational standards.

Duplicate at 4177  ---  Political style will replace merit with election of judges
Duplicate at 4183  ---  Political style will replace merit with election of judges

Duplicate at 4215  ---  Electric car battery technology and costs are improving rapidly
Duplicate at 6356  ---  Electric car battery technology and costs are improving rapidly

Duplicate at 4274  ---  Ending the tourism ban would help undermine the Cuban regime
Duplicate at 4279  ---  Ending the tourism ban would help undermine the Cuban regime

Duplicate at 4425  ---  Torture "to save lives" has a substantial body of support among publics:
Duplicate at 11129  ---  Torture "to save lives" has a substantial body of support among publics:

Duplicate at 4426  ---  "Half of Americans think that torturing terrorist suspects can be "sometimes" or "often" justified."
Duplica

Duplicate at 6157  ---  Exploiting animals is more dignified than their living in the wild
Duplicate at 11768  ---  Exploiting animals is more dignified than their living in the wild

Duplicate at 6163  ---  Humans have a choice and thus responsibility to do no harm to animals.
Duplicate at 11774  ---  Humans have a choice and thus responsibility to do no harm to animals.

Duplicate at 6164  ---  Killing animals should be viewed as equivalent to killing humans
Duplicate at 11775  ---  Killing animals should be viewed as equivalent to killing humans

Duplicate at 6165  ---  Animals should be treated as we would want to be treated
Duplicate at 11776  ---  Animals should be treated as we would want to be treated

Duplicate at 6168  ---  Differences of opinion on killing and hunting must be tolerated
Duplicate at 6208  ---  Differences of opinion on killing and hunting must be tolerated

Duplicate at 6169  ---  Those that disapprove of hunting should not hunt.
Duplicate at 6209  ---  Those

Duplicate at 7206  ---  Kyoto Protocol employs inefficient government controls
Duplicate at 7210  ---  Kyoto Protocol employs inefficient government controls

Duplicate at 7229  ---  Deterrence prevents North Korea from invading South Korea (not mines)
Duplicate at 8308  ---  Deterrence prevents North Korea from invading South Korea (not mines)

Duplicate at 7231  ---  North Korea's tunnel network under DMZ undermines landmines.
Duplicate at 8310  ---  North Korea's tunnel network under DMZ undermines landmines.

Duplicate at 7235  ---  Landmines inhibit a counter-attack into North Korea
Duplicate at 8314  ---  Landmines inhibit a counter-attack into North Korea

Duplicate at 7238  ---  Defending against North Korean blitzkrieg requires landmines
Duplicate at 8317  ---  Defending against North Korean blitzkrieg requires landmines

Duplicate at 7240  ---  Landmines in Korea would force attackers into vulnerable territory.
Duplicate at 8319  ---  Landmines in Korea would force attackers 

Duplicate at 8239  ---  The Pentagon is preparing for war.
Duplicate at 8242  ---  The Pentagon is preparing for war.

Duplicate at 8296  ---  Geneva Conventions regulates use of landmines; ban is excessive.
Duplicate at 8330  ---  Geneva Conventions regulates use of landmines; ban is excessive.

Duplicate at 8493  ---  Multiculturalism highlights differences and can increase hate crimes.
Duplicate at 8502  ---  Multiculturalism highlights differences and can increase hate crimes.

Duplicate at 8540  ---  Economic liberalisation doesn't necessarily cause democratic progression
Duplicate at 8555  ---  Economic liberalisation doesn't necessarily cause democratic progression

Duplicate at 8589  ---  Private banks are better at allocating loans than government
Duplicate at 8592  ---  Private banks are better at allocating loans than government

Duplicate at 8645  ---  Emissions from natural gas can be filtered out
Duplicate at 8690  ---  Emissions from natural gas can be filtered out

Dupl

Duplicate at 9051  ---  Open primaries let independents/third-party-voters participate
Duplicate at 9058  ---  Open primaries let independents/third-party-voters participate

Duplicate at 9138  ---  Regional and local government and security apparatuses would be unable to maintain order following a partitioning of Iraq:
Duplicate at 9147  ---  Regional and local government and security apparatuses would be unable to maintain order following a partitioning of Iraq:

Duplicate at 9235  ---  Debate:Wind energy
Duplicate at 9240  ---  Debate:Wind energy

Duplicate at 9237  ---  Wind energy production is inconsistent relative to demand
Duplicate at 12090  ---  Wind energy production is inconsistent relative to demand

Duplicate at 9250  ---  Debate: Natural gas vehicles
Duplicate at 9256  ---  Debate: Natural gas vehicles

Duplicate at 9272  ---  Polygamous men cannot reciprocate equally the love they receive from their wives.
Duplicate at 9295  ---  Polygamous men cannot reciprocate equall


Duplicate at 11176  ---  Terrorist trials threaten safety of surrounding community
Duplicate at 11225  ---  Terrorist trials threaten safety of surrounding community

Duplicate at 11180  ---  Trying terrorists risks releasing intelligence, costing lives
Duplicate at 11229  ---  Trying terrorists risks releasing intelligence, costing lives

Duplicate at 11181  ---  Courtroom rants of terrorists only expose them as hateful fools
Duplicate at 11230  ---  Courtroom rants of terrorists only expose them as hateful fools

Duplicate at 11186  ---  Trials of 9/11 terrorists in NYC will become a circus
Duplicate at 11235  ---  Trials of 9/11 terrorists in NYC will become a circus

Duplicate at 11187  ---  Trying terrorists shows confidence in US system and ideals
Duplicate at 11236  ---  Trying terrorists shows confidence in US system and ideals

Duplicate at 11188  ---  Military tribunals are a propaganda win for terrorists
Duplicate at 11237  ---  Military tribunals are a propaganda win for t

In [4]:
print(len(duplicateConclusions))

548


In [5]:
df1 = df0.drop(duplicateConclusions)

In [6]:
print(len(df1))

11830


#### 12326 Conclusions -> 11830 Conclusions

### Remove Conclusions that are Urls

In [7]:
i = 0
for conclusion in df1["conclusion"]:
    if re.search(r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+", conclusion):
        print(conclusion)
        print(i)
    i += 1

http://www.populationmedia.org/
2940


In [8]:
df1 = df1.drop([2970])
len(df1)

11829

#### 11830 Conclusions -> 11829 Conclusions

### Remove Square Brackets from Conclusions

In [9]:
for conclusion in df1["conclusion"]:
    i += 1
    if re.search(r"[\[]|[\]]", conclusion):
        print(conclusion)

[[Argument: Circumcision helps prevent herpes Circumcision helps prevent herpes


In [10]:
df1.loc[6484, "conclusion"]\
= "Circumcision helps prevent herpes Circumcision helps prevent herpes"

In [11]:
df1.loc[6484]

argument_ids                                                 7322
frame_ids                                                      75
frame_label                                          adult health
topic_ids                                                     751
topics                                   Infant male circumcision
premise         Edgar J. Schoen, MD, Clinical Professor of Ped...
stances                                                       Yes
conclusion      Circumcision helps prevent herpes Circumcision...
Name: 6484, dtype: object

### Remove Conclusions that are Questions

In [12]:
count = 0
for conclusion in df1["conclusion"]:
    if conclusion[-1] == "?" or conclusion[-2] == "?":
        count += 1
        print(conclusion + "\n")

How is $700b bailout affordable when social programs are not?

Why tax banks and not other bailed-out institutions?

What right does anyone have to deprive another of life on the grounds that they deem that life as not worth living?

Many animals do demonstrate an ability to respect the "rights" of humans?

If humans are animals, why defy our animalistic instincts?

People would never want their own pets tested; why other animals?

If Euthanasia is for 6-month terminal patients, why not let them live it out?

What is more important in Australia than creating a republic?

States have balanced budget amendments, why not US govt?

Why is sexual procreation more "dignified" than scientific procreation?

Will tourists be forbidden from wearing burqa?

Would this ban apply to everyone, or just to candidates?

Why limit speech of corps/unions but not others?

If media can endorse, why can't other comps?

Shareholders aren't concern with media, why w/ other corps?

If gays get civil unions, wh

In [13]:
print(count)

71


In [14]:
for conclusion in df1["conclusion"]:
    if conclusion[-1] == "?" or conclusion[-2] == "?":
        df1 = df1[df1["conclusion"] != conclusion]

In [15]:
print(len(df1))

11758


####  11829 Conclusions -> 11758 Conclusions

### Remove Conclusions that are Topics

In [16]:
count = 0
for conclusion in df1["conclusion"]:
    if len(conclusion) < 15:
        count += 1
        print(conclusion + "\n")

Ron Daniels

Pepper sprays.

Heat hazard.

Zinc hazard:

Algae biofuel

Supervision.

Tooth decay.

Pesticide use.

First step.

Same footing

Obesity risks.

Warren Buffet

Dr Abrahams

No deterrence.

Hilary Clinton

Ted Kennedy

John Kerry:

John McCain

TruthMapping

Iraq Overall:

Free choice.

Deterrence.

No control.

Introduction:

Possible plan:

Andrew Keen

Nicholas Carr



In [17]:
print(count)

27


In [18]:
for conclusion in df1["conclusion"]:
    if len(conclusion) < 15:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11731

#### 11758 Conclusions -> 11731 Conclusions

In [19]:
remove = []
for conclusion in df1["conclusion"]:
    if len(conclusion) < 20:
        remove.append(conclusion)
        print(conclusion + "\n")

Child protection.

Pesticide-free.

Fewer injuries:

Another Dimension:

Too much too soon.

Right incentives.

More dangerous jobs

Organised crime.

Exodus 21:12-14:

Numbers 35:30,31,33

Exodus (20:13):

Dialogue is humble:

Everybody is equal.

Bill Gates, Sr.

Drug Development:

Radiation risk.

Free association

Failing confidence

Rep. Nancy Pelosi

Brookings Institute

Heritage Foundation

AFL-CIO is opposed.

Costs of crime.

Rights of juveniles

Justice is upheld.

Laughter is noisy

Lack of diversity

Home sweet home

Debate:Wind energy

Successful women.

Essential services.

Orwellian future.

God does not exist.

Killing civilians.

Roma are a nation:

Floating currency.

Deeds, not words.

Losing identity.

Practical aspects.

Veal is low in fat

Crime deterrence.

Justice Department

Jürgen Habermas

Transport is safe:



In [20]:
del(remove[43]);del(remove[39]);del(remove[34]);del(remove[32]);del(remove[25]);del(remove[24]);del(remove[21]);del(remove[12]);del(remove[11])
print(len(remove))

35


In [21]:
remove

['Child protection.',
 'Pesticide-free.',
 'Fewer injuries:',
 'Another Dimension:',
 'Too much too soon.',
 'Right incentives.',
 'More dangerous jobs',
 'Organised crime.',
 'Exodus 21:12-14:',
 'Numbers 35:30,31,33',
 'Exodus (20:13):',
 'Bill Gates, Sr.',
 'Drug Development:',
 'Radiation risk.',
 'Free association',
 'Failing confidence',
 'Rep. Nancy Pelosi',
 'Brookings Institute',
 'Heritage Foundation',
 'Costs of crime.',
 'Rights of juveniles',
 'Lack of diversity',
 'Home sweet home',
 'Debate:Wind energy',
 'Successful women.',
 'Essential services.',
 'Orwellian future.',
 'Killing civilians.',
 'Floating currency.',
 'Deeds, not words.',
 'Losing identity.',
 'Practical aspects.',
 'Crime deterrence.',
 'Justice Department',
 'Jürgen Habermas']

In [22]:
for conclusion in df1["conclusion"]:
    if conclusion in remove:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11696

#### 11731 Conclusions -> 11696 Conclusions

In [23]:
remove = []
for conclusion in df1["conclusion"]:
    if len(conclusion) < 25:
        remove.append(conclusion)

In [24]:
remove

['No significant impact.',
 'Lower maintenance costs.',
 'Increased playability.',
 'Other harmful chemicals:',
 'Increased MRSA risk.',
 'We no longer need God.',
 'Dangerous precedent.',
 'Islam is not a race.',
 'Sex toys are unnatural.',
 'getting started tutorial',
 'The ban is hypocritical.',
 'Child labour is hamrful.',
 'The ban in enforceable.',
 'The ban can succeed.',
 'Access to education.',
 'Education is too costly.',
 'Polls are unreliable.',
 'Deuteronomy (30:19):',
 'Dialogue is humble:',
 'Tyranny of minorities.',
 'Everybody is equal.',
 'Charles Schumer (D-NY)',
 'A March, 2005 BNA Poll',
 'Frank Luntz poll 2005',
 'Small business owners:',
 'Getting started tutorial',
 'Geography is irrelevant.',
 'A few notable dates\xa0:',
 'File sharing is illegal.',
 'Free trade is fair trade',
 'Premise of sub-debate.',
 'Radiation risk is small.',
 'Gambling is addictive.',
 'Gambling is regressive',
 'Washington Post 3/24/06',
 'Cato Institute 10/30/06',
 'The White House 1/

In [25]:
len(remove)

78

In [26]:
verbs = [" is "," were "," does "," are ","can ", " need "," tastes "]
rm = []
for conclusion in remove:
    check = False
    for verb in verbs:
        if verb in conclusion:
            rm.append(False)
            check = True
            break
    if not check:
        rm.append(True)
remove = list(compress(remove, rm))
remove

['No significant impact.',
 'Lower maintenance costs.',
 'Increased playability.',
 'Other harmful chemicals:',
 'Increased MRSA risk.',
 'Dangerous precedent.',
 'getting started tutorial',
 'The ban in enforceable.',
 'Access to education.',
 'Deuteronomy (30:19):',
 'Tyranny of minorities.',
 'Charles Schumer (D-NY)',
 'A March, 2005 BNA Poll',
 'Frank Luntz poll 2005',
 'Small business owners:',
 'Getting started tutorial',
 'A few notable dates\xa0:',
 'Premise of sub-debate.',
 'Washington Post 3/24/06',
 'Cato Institute 10/30/06',
 'The White House 1/7/04',
 'Sen John Cornyn (R-TX)',
 'Tom Tancredo (R - Col)',
 'Harming job prospects.',
 'In 2006, Peter Foukal',
 'Cost-benefit analysis.',
 'Cheats keep cheating.',
 'State responsibility.',
 'Rep. Devin Nunes (R-CA)',
 'U.S. Sen. Joe Lieberman',
 'Concentration of wealth.',
 'Political influence.',
 'Negative externalities.',
 'Positive externalities.',
 'Enhanced competition.',
 'Equality in education.',
 'Instant celebrities.',

In [27]:
for conclusion in df1["conclusion"]:
    if conclusion in remove:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11652

#### 11696 Conclusions -> 11652 Conclusions

In [28]:
verbs.append(" apply ")
verbs.append(" speaks ")
verbs.append(" belongs ")
verbs.append(" degrades ")
verbs.append(" should ")
verbs.append(" fosters ")
verbs.append(" tend ")
verbs.append(" costs ")
verbs.append(" appreciate ")
verbs.append(" offer ")
verbs.append(" enhances ")
verbs.append(" have ")
verbs.append(" had ")
verbs.append(" protects ")
verbs.append(" promotes ")
verbs.append(" exist ")
verbs.append(" harms ")
verbs.append(" isn’t ")
verbs.append(" will ")
verbs.append(" do not ")
verbs.append(" do ")
verbs.append(" make ")
verbs.append(" deter ")
verbs.append(" prefer ")
verbs.append(" complains ")
verbs.append(" looks ")
verbs.append(" been ")
verbs.append(" encourage ")
verbs.append(" impairs ")
verbs.append(" lead ")
verbs.append(" helps ")
verbs.append(" shortens ")
verbs.append(" becomes ")
verbs.append(" cannot ")
verbs.append(" seeks ")
verbs.append(" adapts ")
verbs.append(" has ")
verbs.append(" subverts ")
verbs.append(" equals ")
verbs.append(" leads ")
verbs.append(" outweigh ")
verbs.append(" offers ")
verbs.append(" needs ")
verbs.append(" creates ")
verbs.append(" plants ")

In [29]:
remove = []
for conclusion in df1["conclusion"]:
    check = False
    if len(conclusion) < 30:
        for verb in verbs:
            if verb in conclusion:
                check = True
    if not check and len(conclusion) < 30:
        remove.append(conclusion)

In [30]:
print(len(remove))

33


In [31]:
remove

['Dependency on fertilizers.',
 'Many chemicals beyond lead.',
 'Adverse affect on asthmatics.',
 'Professor Tony Blackshield',
 'Difficult to "draw the line".',
 'Risk of further isolation.',
 'No one mentions it anymore.',
 'Appalling working conditions.',
 'Laws of supply and demand.',
 'Poor quality of education.',
 'Debate: Biblical inerrancy',
 'CCTV failing to cut crime',
 'Political barriers exist.',
 'It would strenghten relations',
 'Chuck Hagel (R-Nebraska):',
 'National Immigration Forum',
 'Long-distance friendships.',
 'Protection of the society.',
 'Death penalty for juveniles.',
 'Inequality of comparison.',
 'Acquiring drinking experience',
 'More money for other sectors',
 'Discrimination at workplace.',
 'Rep. Cliff Stearns, R-Florida',
 'Sen. Maria Cantwell (D-WA)',
 'Debate: Natural gas vehicles',
 'Sending the right message.',
 'Giving the right incentives.',
 'More varied points of view.',
 'Lack of collective morale.',
 'Multi-organ retrieval teams.',
 '45.8 mil

In [32]:
for conclusion in df1["conclusion"]:
    if conclusion in remove:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11619

#### 11652 Conclusions -> 11619 Conclusions

In [33]:
remove = []
for conclusion in df1["conclusion"]:
    check = False
    if len(conclusion) < 35:
        for verb in verbs:
            if verb in conclusion:
                check = True
    if not check and len(conclusion) < 35:
        remove.append(conclusion)

In [34]:
print(len(remove))

53


In [35]:
for x, y in zip(remove, range(0, len(remove))):
    print(str(y) + "  ---  " + x + "\n")

0  ---  The spread of procrastination.

1  ---  The rise of cheating and swindle.

2  ---  The right to decide for yourself.

3  ---  Man-made life plays God and nature

4  ---  People survived before cell phones

5  ---  Advertising boosts the economy.

6  ---  Sex toys help fight loneliness.

7  ---  Big govt wrongly to be charitable

8  ---  The Stupid Eggrolls Deserved It

9  ---  CCP's Actions Contradict Marxism

10  ---  If BCS ain't broke, don't fix it

11  ---  Multinationals exploit children

12  ---  Investment in education pays off.

13  ---  Medium- and long-term stability.

14  ---  Avoiding catastrophic mistakes.

15  ---  Costs involved for processing.

16  ---  The EMF could work like the IMF.

17  ---  The Pro-life (anti-abortion) camp

18  ---  Textile and apparel industries

19  ---  Tokamaks could "explode" outward.

20  ---  Cristian Science Monitor 1/27/04

21  ---  District of Columbia v. Heller

22  ---  Facebook undermines stereotypes.

23  ---  Many faiths wel

In [36]:
del(remove[52])
del(remove[49])
del(remove[48])
del(remove[47])
del(remove[46])
del(remove[45])
del(remove[44])
del(remove[40])
del(remove[39])
del(remove[35])
del(remove[34])
del(remove[32])
del(remove[29])
del(remove[28])
del(remove[27])
del(remove[26])
del(remove[22])
del(remove[19])
del(remove[16])
del(remove[12])
del(remove[10])
del(remove[8])
del(remove[6])
del(remove[5])
del(remove[4])
del(remove[3])

In [37]:
print(len(remove))
remove

27


['The spread of procrastination.',
 'The rise of cheating and swindle.',
 'The right to decide for yourself.',
 'Big govt wrongly to be charitable',
 "CCP's Actions Contradict Marxism",
 'Multinationals exploit children',
 'Medium- and long-term stability.',
 'Avoiding catastrophic mistakes.',
 'Costs involved for processing.',
 'The Pro-life (anti-abortion) camp',
 'Textile and apparel industries',
 'Cristian Science Monitor 1/27/04',
 'District of Columbia v. Heller',
 'Many faiths welcome homosexuality.',
 'ICC needed for new global problems',
 'Severe problem, wrong solution.',
 'A House of Representatives vote:',
 '17-year-olds Not Ready to Join',
 'Bias towards the "civilized" west.',
 'The European Court of Human Rights',
 'Downloading troubles consumers.',
 'Natural selection of employees.',
 'Reduction in administrative costs:',
 'Lack of situational awareness.',
 'Possible future inability to act.',
 'Justice Department Defense 1/27/06',
 'Revealing sources and methods:']

In [38]:
for conclusion in df1["conclusion"]:
    if conclusion in remove:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11592

#### 11619 Conclusions -> 11592 Conclusions

In [39]:
notRemove = []
for conclusion in df1["conclusion"]:
    check = False
    if len(conclusion) < 35:
        for verb in verbs:
            if verb in conclusion:
                check = True
    if not check and len(conclusion) < 35:
        notRemove.append(conclusion)

In [40]:
remove = []
for conclusion in df1["conclusion"]:
    check = False
    if len(conclusion) < 40:
        for verb in verbs:
            if verb in conclusion:
                check = True
    if not check and len(conclusion) < 40 and not conclusion in notRemove:
        remove.append(conclusion)

In [41]:
for x, y in zip(remove, range(0, len(remove))):
    print(str(y) + "  ---  " + x + "\n")

0  ---  The deterrent principle still stands.

1  ---  Science increases our quality of life.

2  ---  Anti-adverts send the right message.

3  ---  Countries supporting affirmative action

4  ---  AIG bonuses represent company's greed

5  ---  Terrorists don't fit a neat profile

6  ---  Technology requires time to master.

7  ---  Homeland Security Secretary Tom Ridge

8  ---  A way how to marry for shy persona.

9  ---  Man-made life poses unimaginable risks

10  ---  Police support a ban on assault weapons

11  ---  Replace "communism" with "theocracy"

12  ---  Bailout would be wasted on auto unions

13  ---  Alternatives in emergency situation.

14  ---  "Advertising manipulates small children

15  ---  Cloning treats children as objects:

16  ---  Burqa ban violates religious freedoms

17  ---  Sex toys help fight libido decline.

18  ---  Consumption drops thanks to the ban.

19  ---  Trans fats extend shelf life of foods.

20  ---  Bombing Japan was barbaric and immoral

21  -

In [42]:
notConclusions = []
notConclusions.append(remove[115])
notConclusions.append(remove[100])
notConclusions.append(remove[99])
notConclusions.append(remove[98])
notConclusions.append(remove[91])
notConclusions.append(remove[83])
notConclusions.append(remove[76])
notConclusions.append(remove[67])
notConclusions.append(remove[56])
notConclusions.append(remove[55])
notConclusions.append(remove[48])
notConclusions.append(remove[46])
notConclusions.append(remove[45])
notConclusions.append(remove[44])
notConclusions.append(remove[37])
notConclusions.append(remove[11])
notConclusions.append(remove[7])
notConclusions.append(remove[3])

In [43]:
print(len(notConclusions))
notConclusions

18


['New Mexico Governor Bill Richardson',
 'Wrong incentives for female employees.',
 'Wrong incentives for male employees.',
 'Providing a chance women never had.',
 'Key scholars opposing a partition plan:',
 'Only thing between the gun and the bird',
 'Contradictory arguments by opposition.',
 'More competition for fewer law jobs',
 'Council on Foreign Relations 2/21/06',
 "America for Immigration Control '06",
 'Psychological benefits of gambling:',
 'CNN/USA Today/Gallup poll, August 2005',
 'Actually it went something like this:',
 'Spain vacated her settlement in 1811.',
 'Eternal torture worse than clean break',
 'Replace "communism" with "theocracy"',
 'Homeland Security Secretary Tom Ridge',
 'Countries supporting affirmative action']

In [44]:
for conclusion in df1["conclusion"]:
    if conclusion in notConclusions:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11574

#### 11592 Conclusions -> 11574 Conclusions

In [45]:
for conclusion in df1["conclusion"]:
    check = False
    if len(conclusion) < 45:
        for verb in verbs:
            if verb in conclusion:
                check = True
    if not check and len(conclusion) < 45:
        notRemove.append(conclusion)

In [46]:
remove = []
for conclusion in df1["conclusion"]:
    check = False
    if len(conclusion) < 50:
        for verb in verbs:
            if verb in conclusion:
                check = True
    if not check and len(conclusion) < 50 and not conclusion in notRemove:
        remove.append(conclusion)

In [47]:
for x, y in zip(remove, range(0, len(remove))):
    print(str(y) + "  ---  " + x + "\n")

0  ---  No assurances that a $700b bailout would work

1  ---  $700b bailout includes oversight/transparency

2  ---  Stimulus inadequate to impact massive US economy

3  ---  US stimulus spends too much on roads and bridges

4  ---  Spending got US into crisis, won't get it out

5  ---  Bank tax compensates for damage done by banks

6  ---  AIG bailout benefited banks; bank tax justified

7  ---  Nuclear weapons immorally threaten mass murder

8  ---  Nuclear weapons could fall into terrorists' hands

9  ---  It's not possible to stop nuclear weapons now

10  ---  Many leaders support abolishing nuclear weapons

11  ---  Legal abortions safer than black market abortions

12  ---  Sex-ed teaches nothing new in a culture of sex.

13  ---  Some profit from promoting sexually active teens.

14  ---  Circumcision reduces the risk of prostate cancer.

15  ---  Circumcision reduces the risk of HIV infection

16  ---  Circumcision improves a man's sense of confidence

17  ---  Those circumcis

In [48]:
notConclusions = []
notConclusions.append(remove[42])
notConclusions.append(remove[128])
notConclusions.append(remove[217])
notConclusions.append(remove[264])
notConclusions.append(remove[265])

In [49]:
for conclusion in df1["conclusion"]:
    if conclusion in notConclusions:
        df1 = df1[df1["conclusion"] != conclusion]
len(df1)

11569

#### 11574 Conclusions -> 11569 Conclusions

#### Trying to detect more conclusions that are topics by using a heuristic approach regarding special characters

In [50]:
count = 0
for conclusion in df1["conclusion"]:
    a = conclusion.count(".")
    b = conclusion.count("/")
    c = conclusion.count("(")
    d = conclusion.count(")")
    e = conclusion.count(";")
    f = conclusion.count(",")
    g = conclusion.count(".")
    h = conclusion.count("§")
    
    if (a + b + c + d + e + f + g + h) > 4:
        print(conclusion + "\n")
        count += 1

One of the most famous arguments against abortion is the "dialysis analogy" put forward by Judith Jarvis in 1971. It compares abortion to a situation in which a healthy woman (the mother by analogy) is attached to a dying patient (the fetus by analogy) in order to keep the dying patient alive. The concept is that the dying person does not have a right to the woman's body, and that the woman has a right to "unplug" (abort) even if it means the death of the other person. The problem with the analogy is many fold: 1. A woman and a fetus have a special relationship that is incomparable to that between a woman and a stranger or even a relative. There is a special biological drive inside the mother to keep the baby alive and a dependency by the baby on the mother. The mother, therefore, has a special responsibility to keep her child alive and not abort; 2. A woman often gives a form of tacit approval to the existence of a fetus in her womb: the act of engaging in sexual behavior; 3. abortion

In [51]:
count

93

In [52]:
"Sen. Charles Grassley (R-Iowa), Senate Finance Committee Chairman,"
"Wikipedia:Global warming. Retrieved 11.07.07"
"Joseph Romm. Book Hell and High Water. 2007. pp 12."
"Florida Sen. Bill Nelson (D). \"Florida Deserves a Revote\". March 10, 2008"
"Old memories, old symbols. What's gone, is gone."
"Dan Vermer: Water Privatization. Posted November 14, 2006. - Representing the Coca-Cola Companies water investments"

'Dan Vermer: Water Privatization. Posted November 14, 2006. - Representing the Coca-Cola Companies water investments'

In [53]:
df1 = df1[df1["conclusion"] != "Sen. Charles Grassley (R-Iowa), Senate Finance Committee Chairman,"]
df1 = df1[df1["conclusion"] != "Wikipedia:Global warming. Retrieved 11.07.07"]
df1 = df1[df1["conclusion"] != "Joseph Romm. Book Hell and High Water. 2007. pp 12."]
df1 = df1[df1["conclusion"] != "Florida Sen. Bill Nelson (D). \"Florida Deserves a Revote\". March 10, 2008"]
df1 = df1[df1["conclusion"] != "Old memories, old symbols. What's gone, is gone."]
df1 = df1[df1["conclusion"] != "Dan Vermer: Water Privatization. Posted November 14, 2006. - Representing the Coca-Cola Companies water investments"]
len(df1)

11563

#### 11569 Conclusions -> 11563 Conclusions

#### Remove Conclusions that are whole Arguments

In [54]:
for conclusion in df1["conclusion"]:
    if len(conclusion) > 300:
        print(conclusion + "\n")

One of the most famous arguments against abortion is the "dialysis analogy" put forward by Judith Jarvis in 1971. It compares abortion to a situation in which a healthy woman (the mother by analogy) is attached to a dying patient (the fetus by analogy) in order to keep the dying patient alive. The concept is that the dying person does not have a right to the woman's body, and that the woman has a right to "unplug" (abort) even if it means the death of the other person. The problem with the analogy is many fold: 1. A woman and a fetus have a special relationship that is incomparable to that between a woman and a stranger or even a relative. There is a special biological drive inside the mother to keep the baby alive and a dependency by the baby on the mother. The mother, therefore, has a special responsibility to keep her child alive and not abort; 2. A woman often gives a form of tacit approval to the existence of a fetus in her womb: the act of engaging in sexual behavior; 3. abortion

In [55]:
for conclusion in df1["conclusion"]:
    if len(conclusion) > 300:
        df1 = df1[df1["conclusion"] != conclusion]

In [56]:
len(df1)

11562

#### 11563 Conclusions -> 11562 Conclusions

### Replace Abbreviation in Conclusions

#### Replace "b/w" with "between"

In [57]:
for conclusion in df1["conclusion"]:
    if re.search(r"b/w", conclusion):
        print(conclusion + "\n")

Overlap b/w companies is less efficient than unified govt

Socialism wrongly incites antagonism b/w classes

Competition b/w teachers in charter schools incentivizes performance

Correlation b/w interests of donors and judges doesn't prove corruption.

Diff b/w allowing gay parenting and sanctioning it w/ marriage

Bible does not explicitly define marriage b/w man and woman

Graduated response is best compromise b/w stakeholders

Co-ops create tensions b/w doctors and governing boards

Government research discounts relationship b/w illegality and usage.

Difference b/w encouraging foreign industry abroad and protectionism at home.

Socialism wrongly incites antagonism b/w classes.

"Facebook law" widens gap b/w student lives inside/outside school

There's a desk b/w teachers and students for a reason.



In [58]:
for conclusion in df1["conclusion"]:
    if re.search(r"b/w", conclusion):
        replacement = re.sub(r"b/w", "between", conclusion)
        print(replacement)
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r"b/w", "between", conclusion)

Overlap between companies is less efficient than unified govt
Socialism wrongly incites antagonism between classes
Competition between teachers in charter schools incentivizes performance
Correlation between interests of donors and judges doesn't prove corruption.
Diff between allowing gay parenting and sanctioning it w/ marriage
Bible does not explicitly define marriage between man and woman
Graduated response is best compromise between stakeholders
Co-ops create tensions between doctors and governing boards
Government research discounts relationship between illegality and usage.
Difference between encouraging foreign industry abroad and protectionism at home.
Socialism wrongly incites antagonism between classes.
"Facebook law" widens gap between student lives inside/outside school
There's a desk between teachers and students for a reason.


#### Replace "w/o" with "without"

In [59]:
for conclusion in df1["conclusion"]:
    if re.search(r"w/o", conclusion):
        print(conclusion + "\n")

Arizona law will harass, jail Latinos w/o papers on them

Arizona has right to fight illegal immigration w/o federal authorization

Arizona law constitutional; no stops w/o other offense.

Trans fats can be replaced w/o changing taste/price.

Civil unions give gays equal benefits w/o changing marriage laws

Nuclear weapons, deterrence can be maintained w/o detonation tests

Incentive to learn English exists w/o making it official language

Bilingual programs allow students to get by w/o learning English

Lisbon treaty amends EU/EC treaties w/o replacing them

Balanced left/right broadcasting exist w/o Fairness Doctrine

Gays can have stable relationships w/o marriage

Gays raise children now, but at a disadvantage w/o marriage

Unethical to change planet w/o universal approval

Graduated response relies on mere accusations w/o due process.

Developing states need room to develop w/o adaptation burdens.

Casualties in repelling N. Korean invasion would be higher w/o mines.

US and other

In [60]:
for conclusion in df1["conclusion"]:
    if re.search(r"w/o", conclusion):
        replacement = re.sub(r"w/o", "without", conclusion)
        print(replacement)
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r"w/o", "without", conclusion)

Arizona law will harass, jail Latinos without papers on them
Arizona has right to fight illegal immigration without federal authorization
Arizona law constitutional; no stops without other offense.
Trans fats can be replaced without changing taste/price.
Civil unions give gays equal benefits without changing marriage laws
Nuclear weapons, deterrence can be maintained without detonation tests
Incentive to learn English exists without making it official language
Bilingual programs allow students to get by without learning English
Lisbon treaty amends EU/EC treaties without replacing them
Balanced left/right broadcasting exist without Fairness Doctrine
Gays can have stable relationships without marriage
Gays raise children now, but at a disadvantage without marriage
Unethical to change planet without universal approval
Graduated response relies on mere accusations without due process.
Developing states need room to develop without adaptation burdens.
Casualties in repelling N. Korean inva

In [61]:
for conclusion in df1["conclusion"]:
    if re.search(r"W/o", conclusion):
        print(conclusion + "\n")

W/o mandates insurers focus on risk instead of costs

W/o mines more US troops would be needed in South Korea.

W/o mines more US troops would be needed in South Korea

W/o more troops, Afgh will become terrorist haven

W/o Net Neutrality, price discrimination risks stifling start-ups

W/o net neutrality Internet has fast and slow lanes

W/o public option, US health care law emboldens insurers



In [62]:
for conclusion in df1["conclusion"]:
    if re.search(r"W/o", conclusion):
        replacement = re.sub(r"W/o", "Without", conclusion)
        print(replacement)
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r"W/o", "Without", conclusion)

Without mandates insurers focus on risk instead of costs
Without mines more US troops would be needed in South Korea.
Without mines more US troops would be needed in South Korea
Without more troops, Afgh will become terrorist haven
Without Net Neutrality, price discrimination risks stifling start-ups
Without net neutrality Internet has fast and slow lanes
Without public option, US health care law emboldens insurers


#### Replace "w/" with "with"

In [63]:
for conclusion in df1["conclusion"]:
    if re.search(r" w/", conclusion):
        print(conclusion + "\n")

Amend would create Const. crisis, court battle w/ each budget.

18% spending rule would not fit w/ other timeless Amendments.

Reducing spending w/ BBA frees-up money for job-creation.

Politicians will favor corporate interests w/ balanced budget.

Suicide bombers don't care to disguise identity w/ burqas.

The right to free movement only applies w/in borders.

Diff between allowing gay parenting and sanctioning it w/ marriage

Opposing ground zero mosque supports message of US at war w/ Islam.

Cease-fire invalideates blockade-justification of "war" w/ Hamas

Cease-fire invalidates blockade-justification of "war" w/ Hamas

Liberals lost in deal w/ no tax increases, sharp cuts.



In [64]:
for conclusion in df1["conclusion"]:
    if re.search(r" w/", conclusion):
        replacement = re.sub(r" w/", " with", conclusion)
        print(replacement)
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r" w/", " with", conclusion)

Amend would create Const. crisis, court battle with each budget.
18% spending rule would not fit with other timeless Amendments.
Reducing spending with BBA frees-up money for job-creation.
Politicians will favor corporate interests with balanced budget.
Suicide bombers don't care to disguise identity with burqas.
The right to free movement only applies within borders.
Diff between allowing gay parenting and sanctioning it with marriage
Opposing ground zero mosque supports message of US at war with Islam.
Cease-fire invalideates blockade-justification of "war" with Hamas
Cease-fire invalidates blockade-justification of "war" with Hamas
Liberals lost in deal with no tax increases, sharp cuts.


### Replace Prefixes like "Argument:" or "Debate:"

In [65]:
for conclusion in df1["conclusion"]:
    if re.search(r"^.+:", conclusion):
        if (len(re.search(r"^.+:", conclusion).group(0))) < 30:
            print(conclusion + "\n")

Romans 4:9-12 supports the practice of circumcision.

Position:The Urban League supports Affirmative Action

Carbon Emissions: Market vs. Regulatory Approaches (arguments)

Carbon Emissions: Cap-and-trade versus Carbon Tax (arguments)

Moral hazard: US auto bailout would reward failure

Energy security: cellulosic ethanol reduces dependence on foreign oil.

Argument: Obama can win the "big" states in the general election

Dialogue is humble:

This law protects evil doers:

Argument: Earmarks enable strong congressional check on executive spending

Debate: Graduated response antipiracy laws

High-speed trains: costly govt project in search of need

Myth: circumcision is necessary later if not done at birth

Wikipedia article: "Scientific opinion on Global Climate Change"

Sado-masochism is dangerous:

Argument: Jyllands-Posten cartoons have led to the curtailment of free speech

Argument: Religious people must be tolerant of criticism in a modern democracy

Argument: Muslims should not 

In [66]:
prefixes = [
    "Position:",
    "Carbon Emissions:",
    "Moral hazard:",
    "Energy security:",
    "Argument:",
    "Debate:",
    "High-speed trains:",
    "Myth:",
    "Wikipedia article:",
    "The three D's:"
]

In [67]:
for conclusion in df1["conclusion"]:
    for prefix in prefixes:
        if prefix in conclusion:
            print(conclusion.replace(prefix, "") + "\n")
            df1.loc[df1["conclusion"] == conclusion, "conclusion"] = conclusion.replace(prefix, "")

The Urban League supports Affirmative Action

 Market vs. Regulatory Approaches (arguments)

 Cap-and-trade versus Carbon Tax (arguments)

 US auto bailout would reward failure

 cellulosic ethanol reduces dependence on foreign oil.

 Obama can win the "big" states in the general election

 Earmarks enable strong congressional check on executive spending

 Graduated response antipiracy laws

 costly govt project in search of need

 circumcision is necessary later if not done at birth

 "Scientific opinion on Global Climate Change"

 Jyllands-Posten cartoons have led to the curtailment of free speech

 Religious people must be tolerant of criticism in a modern democracy

 Muslims should not be given exceptional rights and protections

 Natural gas can be harmful and even poisonous to ingest

 The Three Gorges Dam has destroyed Chinese cultural artifacts

 The UN veto is anachronistic

 oil rigs drill, drain and devastate.



#### Remove again a few Conclusion that are no actual Conclusions

In [68]:
remove = [
    " Market vs. Regulatory Approaches (arguments)",
    " Cap-and-trade versus Carbon Tax (arguments)",
    " Graduated response antipiracy laws",
    " costly govt project in search of need",
    " \"Scientific opinion on Global Climate Change\"",
    "\"Tibet: Its Ownership and Human Rights Situation,\" also known as the \"China White Paper\". Issued by Information Office of the State Council of the People's Republic of China in September 1992. (The definitive Chinese Government line on Tibet)",
]

In [69]:
for conclusion in df1["conclusion"]:
    if conclusion in remove:
        df1 = df1[df1["conclusion"] != conclusion]
print(len(df1))

11556


#### 11562 Conclusions -> 11556 Conclusions

### Remove Text in Brackets

In [70]:
for conclusion in df1["conclusion"]:
    if re.search(r"\([^\)]+\)", conclusion):
        print(conclusion + "\n")

Planed breaks in the 700 mile fence are filled by mountainous terrain (sufficient barriers).

Denying abortion rights forces maternity on women (state rape)

The right to choice/privacy (abortion) does not override the right to life

Wanted fetuses are beloved "babies"; unwanted ones are "tissue" (inconsistent)

If women (not men) are solely burdened by pregnancy, they must have a choice.

Abortion is just when birth control fails (involuntary impregnation)

The Declaration of Independence states we have a right to Life (Liberty and the Pursuit of Happiness).

Many teens don't have (responsible) parents; schools must teach sex-ed.

Mutual assured destruction (nuclear deterrence).

AIG signed bonus contracts as golden parachutes (invalid).

AIG signed bonus contracts on assurances of bailout (invalid).

AIG knows best how to spend bailout (even on bonuses)

Algae biofuel requires substantial land (expensive)

Cruelty to animals (e.g. bull fighting, fox hunting, battery hen farming) is t

In [71]:
for conclusion in df1["conclusion"]:
    if re.search(r"\([^\)]+\)", conclusion):
        print(re.sub(r"\([^\)]+\)", "", conclusion) + "\n")
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r"\([^\)]+\)", "", conclusion)

Planed breaks in the 700 mile fence are filled by mountainous terrain .

Denying abortion rights forces maternity on women 

The right to choice/privacy  does not override the right to life

Wanted fetuses are beloved "babies"; unwanted ones are "tissue" 

If women  are solely burdened by pregnancy, they must have a choice.

Abortion is just when birth control fails 

The Declaration of Independence states we have a right to Life .

Many teens don't have  parents; schools must teach sex-ed.

Mutual assured destruction .

AIG signed bonus contracts as golden parachutes .

AIG signed bonus contracts on assurances of bailout .

AIG knows best how to spend bailout 

Algae biofuel requires substantial land 

Cruelty to animals  is the sign of an uncivilized society.

The ends  don't justify the means 

Cells and other surrogates cannot be used in the case of behavioral experiments 

Anorexics are typically treated under mental health legislation :

Some doctors advocate focusing on palliati

### Replace weird Quotation Marks

In [93]:
weirdQutationMarks = ["“", "”", "’", "‘"]–`

SyntaxError: invalid character in identifier (<ipython-input-93-ee8adb2f51b9>, line 1)

In [94]:
for conclusion in df1["conclusion"]:
    if re.search(r"“[^”]+”", conclusion):
        print(conclusion + "\n")

No one argues that an acorn or even an animal fetus has a “sanctity” or rights.

Regarding the atheists’ premise “If you believe in God, then you reject reason.”

Institutions of global governance mean a “democratic deficit,”

As higher earners, graduates already pay a lot more on average in taxes over their lifetime, while consuming less in welfare payments, thus more than repaying their “debt to society”:

Philip Martin of UC Davis and Michael Teitelbaum, “The Mirage of Mexican Guest Workers,” Nov/Dec 2001 Foreign Affairs.

“Backloading” of the primary schedule can help reduce the special interest effect contained in all “big money ‘ campaigns.



In [95]:
for conclusion in df1["conclusion"]:
    if re.search(r"“[^”]+”", conclusion):
        print(re.sub(r"“|”", "\"", conclusion))
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r"“|”", "\"", conclusion)

No one argues that an acorn or even an animal fetus has a "sanctity" or rights.
Regarding the atheists’ premise "If you believe in God, then you reject reason."
Institutions of global governance mean a "democratic deficit,"
As higher earners, graduates already pay a lot more on average in taxes over their lifetime, while consuming less in welfare payments, thus more than repaying their "debt to society":
Philip Martin of UC Davis and Michael Teitelbaum, "The Mirage of Mexican Guest Workers," Nov/Dec 2001 Foreign Affairs.
"Backloading" of the primary schedule can help reduce the special interest effect contained in all "big money ‘ campaigns.


In [96]:
for conclusion in df1["conclusion"]:
    if re.search(r"‘[^’]+’", conclusion):
        print(conclusion + "\n")

Arranged marriages are very much ‘real’ marriages.

Arranged marriage is not a true ‘cultural value’ that is in some sense inviolate.

Factory farming sees animals as ‘products,’ ‘commodities’ for production and sale instead of as life forms that deserve a certain respect and good treatment:

Within democracies, voters are entitled to know the ‘down-side’ to a candidate, and the other side is obviously well placed to voice it:

The concept of ‘rights’ is problematic when multiple rights are infringed.

‘Guilt’ is not verifiable in the same way in a second trial.

If the ‘double jeopardy’ rule is scrapped, police work will be sloppier

Wimbledon stands out as discriminatory given that two other ‘Grand Slam’ tournaments  pay equal prize money.

Revenue from a ‘fat tax’ could be used toward health programs.

To talk of an ‘Islamic democracy’ is to distort the concept of democracy to an unrecognisable extent.

Legalization of prostitution would disrupt the corruption of ‘pimps’.

The propo

In [98]:
for conclusion in df1["conclusion"]:
    if re.search(r"‘[^’]+’", conclusion):
        print(re.sub(r"‘|’", "\"", conclusion))
        df1.loc[df1["conclusion"] == conclusion, "conclusion"] = re.sub(r"‘|’", "\"", conclusion)

Arranged marriages are very much "real" marriages.
Arranged marriage is not a true "cultural value" that is in some sense inviolate.
Factory farming sees animals as "products," "commodities" for production and sale instead of as life forms that deserve a certain respect and good treatment:
Within democracies, voters are entitled to know the "down-side" to a candidate, and the other side is obviously well placed to voice it:
The concept of "rights" is problematic when multiple rights are infringed.
"Guilt" is not verifiable in the same way in a second trial.
If the "double jeopardy" rule is scrapped, police work will be sloppier
Wimbledon stands out as discriminatory given that two other "Grand Slam" tournaments  pay equal prize money.
Revenue from a "fat tax" could be used toward health programs.
To talk of an "Islamic democracy" is to distort the concept of democracy to an unrecognisable extent.
Legalization of prostitution would disrupt the corruption of "pimps".
The proposed system 

In [101]:
for conclusion in df1["conclusion"]:
    if re.search(r"‘", conclusion):
        print(conclusion + "\n")

"Backloading" of the primary schedule can help reduce the special interest effect contained in all "big money ‘ campaigns.



In [102]:
for conclusion in df1["conclusion"]:
    if re.search(r"’", conclusion):
        print(conclusion + "\n")

A healthier body weight is necessary to be able to treat the patient’s psychological problems:

An anorexic’s fear of weight gain, especially forced weight gain in hospital is an obstacle to treatment:

Allowing Burma to join ASEAN seemed to signal approval of that regime’s record on human rights:

Regarding the atheists’ premise "If you believe in God, then you reject reason."

The desire to have one’s own child and to nurture it is wholly natural:

It is more important to protect people’s health than to protect businesses.

Israel’s security wall was found by the ICJ to be a breach of international law in 2004

Even the Israeli High Court has called upon Israel’s government, that it should not provide security to Israeli settlers at the expense of Palestinian rights

The barrier harms Israel’s international image and weakens its support in the USA, a crucial ally

The barrier does not harm Israel’s international position

The reassurance which comes from knowing one’s parentage is a 

In [104]:
for conclusion in df1["conclusion"]:
    if re.search(r"–", conclusion):
        print(conclusion + "\n")

The market decides the ratio of reward – and there isn’t a fairer method of deciding than that:



In [106]:
for conclusion in df1["conclusion"]:
    if conclusion[-1] == ":" or conclusion[-2] == ":":
        print(conclusion + "\n")

A fetus is not a "person" so can't have rights protecting it from abortion:

Abortion deprives a fetus of an entire human future:

Post-abortion syndrome is not a medically recognized syndrome:

A child should not be killed to save a mother:

Abortion is justified when the fetus is certain to suffer and die from a disability:

The fact that a child is likely to have a short life does not justify further shortening it:

A ban on abortion presents practical problems of enforcement:

Difficulties of enforcement should not diminish the principles of the law:

The reproductive rights movement has no genocidal drive:

Affirmative action lowers value of degrees earned by minorities:

Anorexics are typically treated under mental health legislation :

Medical ethics say that a doctor has a responsibility to keep the patient alive to administer treatment:

Force feeding is undignified and thus illegal:

A healthier body weight is necessary to be able to treat the patient’s psychological problems